shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -241,4 +230,50 @@ internal class StepQuizActionDispatcher(
                     StepQuizFeature.SubmissionState.Loaded(submission)
                 }
             }
+
+    private suspend fun handleUpdateProblemsLimitAction(
+        action: InternalAction.UpdateProblemsLimit,
+        onNewMessage: (Message) -> Unit
+    ) {
+        if (!currentProfileStateRepository.isFreemiumWrongSubmissionChargeLimitsEnabled()) return
+
+        freemiumInteractor.chargeProblemsLimits(action.chargeStrategy)
```

Comment:
Why problems limit charge is placed after `isFreemiumWrongSubmissionChargeLimitsEnabled` check?
As I see, `chargeProblemsLimits` check this condition internally.
I understand, that the only possible value for `action.chargeStrategy` is `AFTER_WRONG_SUBMISSION`, but there is a change to pass `AFTER_CORRECT_SUBMISSION`.

Reply:
`InternalAction.UpdateProblemsLimit` must be send only when `freemium.wrong_submission_charge_limits` feature enabled, but it was inconvenient to do that check in the reducer.

So, here this is early exit, and to not produce unnecessary new message.